content/rest-api/index.textile

```diff
@@ -149,6 +169,67 @@ In principle the link URL might be any valid URL but in practice it will always
 
 Clients that are unable to process response headers may also request an @envelope@ response type.
 
+h3. Control of response content
+
+By default the response body of a query response will contain the requested resource, encoded in the requested format. However, there are also query params that allow the response body to be adjusted to contain only a subset of the representation, or a restructured representation.
+
+The following params are supported:
+
+h4. fields
+
+Specifying @?fields=<field spec>[,<field spec>, ...]@ returns the representation containing only the specified fields. A field spec is a specifier which is either a single field - in which case the representation contains only that fields's value - or a dot-separated sequence of fields, in which case the representation contains only the fields matching those given in the field spec at each level in the hierarchy.
+
+h5. Example
+
+```[sh]
+  curl https://rest.ably.io/stats?fields=channels.peak,intervalId \
+       -u "{{API_KEY}}"
+
+  # Response
+  [{
+      "channels": { "peak": 2 },
+      "intervalId": "2015-11-20:15"
+  }]
+```
+
+h4. flatten
+
+Specifying @?flatten=true@ will result in a flattened representation, with the returned object structure being flattened into one with a single level of long keys instead of a deep structure. When the results are contained in an array, each result key is prefixed with an index in the form @[index].[key]@
+
+h5. Example
+
+```[sh]
+  curl https://rest.ably.io/stats?flatten=true&fields=channels.peak,intervalId \
+       -u "{{API_KEY}}"
+
+  # Response
+  {
+      "0.channels.peak": 2,
+      "0.intervalId": "2015-11-20:15"
+  }
+```
+
+h4. select
+
+Specifying @?select=<path spec>@ returns a representation of the resource containing only the fields that match the given path specification. Like the fields param this permits only specific fields to be obtained, but it the path spec may contain wildcard elements, and all matching fields in the representation are returned.
+
+The result is a partially-flattened representation, as a map whose keys are the long keys that match the path spec, and the values are the corresponding values. When the results are contained in an array, each result key is prefixed with an index in the form @[index].[key]@
```

Comment:
> When the results are contained in an array

This needs to be worded in a way that applies to arrays at all levels, not just the top-level.

When the results contain array elements, the array index of each element is included as the corresponding component of the result key.